### DISCUSSION TITLE
[FEATURE REQUEST] Suggestion to add new buttons and new features to the toolbar

### DISCUSSION THREAD

**Discussion by universish** [zen-browser/desktop#2866] [Ideas]
Can you add “Activate compact mode”, “Reopen closed tab” and “Close this open tab”? Please...

1. Compact mode, I turn it on by right-clicking on the toolbar > click on “compact mode” > click on “activate compact mode”. This is difficult because I activate and disable it so often. I would like to do this with one button.

2. I really want the “Close this open tab” feature, because when the “Expand Sidebar” button (the icon with the top arrow to the right and the bottom arrow to the left) is closed, the logos of the open tabs appear. Their names are not written. The “close tab” icon in the shape of an “x” does not appear either. I have to close the tab by hovering over it > Right click > Click close tab, which is a pain when I want to close many tabs.

When the “Expand Sidebar” button is on, the logos and names of the tabs appear, and the “Close tab” icon (x) appears when you mouse over it. But when the “Expand Sidebar” button is on, it takes up a lot of space on my screen, even though I have shortened it, it takes up a lot of space. the reduction of screen space makes it difficult to use on a laptop.

So it would be very useful if a “Close this open tab” button could be created or improved (there is no such button in zen browser yet, only “Close tab”). If a “Close tab” button is made, there may be errors in the system. It may not know which tab to close. That's why I suggest creating a “Close this open tab” button.

3. I also use the “Reopen closed tab” button a lot. Because my brain works differently, I think I'm done with a tab and close it immediately; then I realize I'm not done and reopen many tabs. I do this a lot. I have to go through the steps Tabs > right click > left click “Reopen closed tab” button. If this feature is available, it will make my work a lot easier.

Note: We add the buttons to the toolbar with these steps:
Sidebar > Right click > Customize toolbar > drag icon > drop to toolbar

**Comment by universish** [comment]
Can toolbar buttons with these features be distributed from the mod store?
Or do these features need to be added to zen-browser desktop via update?

**Comment by universish** [comment]
@LeMoonStar @IAmJafeth @HarryHeres @mr-cheff @n7itro @NOCanoa @kristijanribaric @BrhmDev @larziwau @omove @JosueGalRe @vraravam @Muko-Tabi @mohvn @DanielSRS @shvhil @notsatvrn @katedickey @Richard-Woessner @neurokitti @Muko-Tabi @AlternateRT @NOCanoa

Let's add toolbar customization buttons in zen mods for zen browser. This way, those who want to create a toolbar button can upload it to zen mods and add it.
Or we can distribute these toolbar buttons embedded in zen-desktop.

Such toolbar buttons and shortcuts will increase browser efficiency. Please consider adding the 3 toolbar buttons I wrote in this thread to zen browser desktop.
Sincerely regards

**Reply by mr-cheffy** [reply]
Why do you keep pinging random people

**Comment by n7itro** [comment]
Ideally Zen Mods should support JS, which solves this issue

**Comment by larziwau** [comment]
@mr-cheff people like this should get banned from the repo. this is not needed and devs will look onto this later. theyre just busy

**Reply by mr-cheffy** [reply]
Done

**Reply by larziwau** [reply]
thanks. anyway i think we should make something like an action to warn people when pinging devs. i can look onto that if you want

**Comment by universish** [comment]
@larziwau are u dictator?

**Reply by nocanoa** [reply]
Pinging everybody is not cool! 
And it's hell annoying
